CMakeLists.txt

```diff
@@ -391,6 +444,40 @@ set(WOLFSSL_SLOW_MATH "yes")
 #       - Public key callbacks
 #       - Microchip/Atmel CryptoAuthLib
 
+# lean psk build
+add_option("WOLFSSL_LEAN_PSK"
+    "Enable Lean PSK build (default: disabled)"
+    "no" "yes;no")
+
+if(${WOLFSSL_LEAN_PSK})
+    list(APPEND WOLFSSL_DEFINITIONS "-DWOLFSSL_LEANPSK -DWOLFSSL_STATIC_PSK -DHAVE_NULL_CIPHER -DSINGLE_THREADED -DNO_AES -DNO_FILESYSTEM -DNO_RSA -DNO_DSA -DNO_DH -DNO_PWDBASED -DNO_MD4 -DNO_MD5 -DNO_ERROR_STRINGS -DNO_OLD_TLS -DNO_RC4 -DNO_WRITEV -DNO_DEV_RANDOM -DWOLFSSL_USER_IO -DNO_SHA")
+    if(WOLFSSL_SLOW_MATH)
+        set(WOLFSSL_SLOW_MATH "no")
+    endif()
+    if(WOLFSSL_SINGLE_THREADED)
```

Comment:
The variable ENABLED_SLOWMATH gets set in configure.ac but it doesn't seem to get used there.

Reply:
Probably it would be better to consolidate all options in the first part of the CMake file and then put all logic for the overrides below the options?

Reply:
> Probably it would be better to consolidate all options in the first part of the CMake file and then put all logic for the overrides below the options?

Definitely. It's just a rather large undertaking to do that, and I unfortunately don't have time to work on it.